README.md

```diff
@@ -1,135 +1,221 @@
+**Note: This guide is intended to be used by developers. If you're not a developer - please find one and come back later :)**
+
+Metrilo helps you track the events that the people visiting your shop do - such as viewing specific pages or products, adding products to the cart, making orders, etc. The provided data is transformed into different reports and analytics metrics you could benefit from.
+
+For more information about Metrilo and its features please visit [the official website](https://www.metrilo.com/).
+
+## Integration overview
+
+:warning: **We strongly recommend doing this on a Staging/Beta environment first to make sure the integration is smooth and nothing breaks down.**
+
+Here is what you need to do in order to integrate your data with Metrilo:
+
+1. Navigate to the `Settings -> Installation` page in your [Metrilo project](https://app.metrilo.com) (if you don't already have one, you can [create one here](https://app.metrilo.com/signup) and note your `API Token` - you'll need it for the next two steps
+2. [Import your resources](#importing-resources) to Metrilo
+3. [Install the tracking library](#tracking-library-installation) on each page of your website that you want to track
+4. Send [tracking events](#sending-tracking-events) to Metrilo for each user action on your website
+
+That's it! Now user actions on your website will be recorded in your [Metrilo project](https://app.metrilo.com).
+
+## API
+
+Our API documentation is written according to the [OpenAPI Specification](https://swagger.io/docs/specification/about/). You can find all of the endpoints the Metrilo API serves and what parameters they require [here](https://app.swaggerhub.com/apis/metrilo/api/1.0.0).
+
+All calls to these endpoints have to be done from your backend - therefore, we don't provide any specific code examples as the implementation is bound to your backend logic and programming language.
+
+:warning: Each Metrilo project has its own `API Token` and multiple stores cannot point to the same Metrilo project.
+
+### Importing resources
+
+A _resource_ is a **category**, **product**, **customer** or **order**. Any time one of these changes on your backend, you should call the respective endpoint to create or update the resource in Metrilo.
+
+Before sending any [tracking events](#sending-tracking-events) to Metrilo, you need to import your data using the endpoints provided for each resource. All of the endpoints have a singular and a batch version - we recommend using the latter to import your resources. You can find more detailed information about what data they require in the [documentation](https://app.swaggerhub.com/apis/metrilo/api/1.0.0).
+
+Importing **must** be done in the following order:
+1. Categories - `/category/batch` _(required)_
+2. Products `/product/batch` _(required)_
+3. Customers `/customer/batch` _(required)_
+4. Orders - `/order/batch` (required _only if you want to sync any historical data with Metrilo_)
```

Comment:
Not sure where the import order comes from but in the plugins the import order is coded in the following way:
1. Customer
2. Categories
3. Deleted Products
4. Products
5. Orders

Not sure if products import step is accounting for products present in old orders but removed from current shop nomenclature ?

Reply:
Customers can be done before or after products - the only important thing is that it's done _before_ orders - but I don't think we should complicate that for people.

Regarding old products - I will make a note of that somewhere

Reply:
@nedick007  I'll change it to `Customers` first so that it's consistent